test/unit/lib/plugins/aws/package/compile/functions.test.js

```diff
@@ -1756,6 +1756,271 @@ describe('lib/plugins/aws/package/compile/functions/index.test.js', () => {
         expect(fooFunctionRole).to.equal(fooFunctionConfig.role);
       });
     });
+
+    it('should fail if `functions[].image` references image without path and uri', async () => {
+      await expect(
+        runServerless({
+          fixture: 'function',
+          cliArgs: ['package'],
+          configExt: {
+            provider: {
+              docker: {
+                images: {
+                  invalidimage: {},
+                },
+              },
+            },
+            functions: {
+              fnProviderInvalidImage: {
+                image: 'invalidimage',
+              },
+            },
+          },
+        })
+      ).to.be.rejectedWith(
+        'Either "uri" or "path" property needs to be set on image "invalidimage"'
+      );
+    });
+
+    it('should fail if `functions[].image` references image with both path and uri', async () => {
+      await expect(
+        runServerless({
+          fixture: 'function',
+          cliArgs: ['package'],
+          configExt: {
+            provider: {
+              docker: {
+                images: {
+                  invalidimage: {
+                    path: './',
+                    uri:
+                      '000000000000.dkr.ecr.sa-east-1.amazonaws.com/test-lambda-docker@sha256:6bb600b4d6e1d7cf521097177dd0c4e9ea373edb91984a505333be8ac9455d38',
+                  },
+                },
+              },
+            },
+            functions: {
+              fnProviderInvalidImage: {
+                image: 'invalidimage',
+              },
+            },
+          },
+        })
+      ).to.be.rejectedWith(
+        'Either "uri" or "path" property (not both) needs to be set on image "invalidimage"'
+      );
+    });
+
+    it('should fail if `functions[].image` references image from `provider.docker.images` that has invalid path', async () => {
+      await expect(
+        runServerless({
+          fixture: 'docker',
+          cliArgs: ['package'],
+          configExt: {
+            provider: {
+              docker: {
+                images: {
+                  baseimage: {
+                    path: './invalid',
+                  },
+                },
+              },
+            },
+          },
+        })
+      ).to.be.rejectedWith('Could not access Dockerfile under path: "invalid/Dockerfile"');
+    });
+
+    describe('with `provider.docker.images` that require building', () => {
```

Comment:
My gut feeling is to have a test case for every scenario here (so every possible error thrown by `spawnExt`), but these tests require separate `runServerless` - do you agree? The same for `deployFunction.test.js` scenarios

Reply:
> but these tests require separate runServerless - do you agree?

Yes, unfortunately it's case, and I don't think we can workaround that.

_As a side note: We will have to optimize AJV due to how slowly it setups with v7 (I will shortly describe my proposal in an issue) that will also ensure that such repeated `runServerless` runs run faster_